src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -356,20 +356,31 @@ module _ {ℓ} {R : Rel A ℓ} (R? : Decidable R) where
   ... | false | _   | rec = begin
     y ∷ merge R? (x ∷ xs) ys <⟨ rec ⟩
     y ∷ x ∷ xs ++ ys         ↭⟨ shift y (x ∷ xs) ys ⟨
-    (x ∷ xs) ++ y ∷ ys       ≡⟨ Lₚ.++-assoc [ x ] xs (y ∷ ys) ⟨
+    (x ∷ xs) ++ y ∷ ys       ≡⟨ List.++-assoc [ x ] xs (y ∷ ys) ⟨
     x ∷ xs ++ y ∷ ys         ∎
     where open PermutationReasoning
 
+------------------------------------------------------------------------
+-- foldr of Commutative Monoid
+
+module _ (commutativeMonoid : CommutativeMonoid a r) where
+
+  open module CM = CommutativeMonoid commutativeMonoid
+
+  foldr-commMonoid : ∀ {xs ys} → xs ↭ ys → List.foldr _∙_ ε xs ≈ List.foldr _∙_ ε ys
```

Comment:
Shouldn't this be more like foldr respects permutation? From the name, I was expecting a definition of foldr for a commutative monoid.

Reply:
*Very* happy to have a better name for this: I had been planning to deprecate the corresponding one under `Setoid.Properties` because it takes an `IsCommutativeMonoid` instance, and I think such things should take *bundles* not *structures*.